substrate/frame/staking/src/lib.rs

```diff
@@ -740,25 +743,35 @@ pub struct Nominations<T: Config> {
 ///
 /// This is useful where we need to take into account the validator's own stake and total exposure
 /// in consideration, in addition to the individual nominators backing them.
-#[derive(Encode, Decode, RuntimeDebug, TypeInfo, PartialEq, Eq)]
-pub struct PagedExposure<AccountId, Balance: HasCompact + codec::MaxEncodedLen> {
+#[derive(Encode, Decode, RuntimeDebug, TypeInfo, PartialEq, Eq, MaxEncodedLen)]
+pub struct PagedExposure<
+	AccountId,
+	Balance: HasCompact + MaxEncodedLen,
+	MaxExposurePageSize: Get<u32>,
+> {
 	exposure_metadata: PagedExposureMetadata<Balance>,
-	exposure_page: ExposurePage<AccountId, Balance>,
+	exposure_page: ExposurePage<AccountId, Balance, MaxExposurePageSize>,
 }
 
-impl<AccountId, Balance: HasCompact + Copy + AtLeast32BitUnsigned + codec::MaxEncodedLen>
-	PagedExposure<AccountId, Balance>
+impl<
+		AccountId,
+		Balance: HasCompact + Copy + AtLeast32BitUnsigned + MaxEncodedLen,
+		MaxExposurePageSize: Get<u32>,
+	> PagedExposure<AccountId, Balance, MaxExposurePageSize>
 {
 	/// Create a new instance of `PagedExposure` from legacy clipped exposures.
 	pub fn from_clipped(exposure: Exposure<AccountId, Balance>) -> Self {
```

Comment:
this method seems no longer used in the codebase, maybe let's change the signature to return `Result<Self, ...>` to be safer. So caller is aware of the constraint.

Reply:
Added in https://github.com/paritytech/polkadot-sdk/pull/6445/commits/c881bb0015cfd0c9248ebd53f751ff267b5ba5d9

Reply:
Yes, but I was thinking instead of doing
```rust
		let others = WeakBoundedVec::try_from(exposure.others).unwrap_or_default();
		defensive_assert!(old_exposures == others.len(), "Too many exposures for a page");
```
we should return an error.